### DISCUSSION TITLE
mcu 'mcu' : Unable to connect.

### DISCUSSION THREAD

**Discussion by Baustin525** [Guilouz/Klipper-Flsun-Speeder-Pad#138] [Q&A]
I have been fighting with this for a few days mow, I am not a novice, but by no means an expert.  
I have updated to printer.cfg with the serial port the command lists for a single instance.
I have tried re-flashing the firmware multiple times, I have the Nano V2.1 board.
I did have a problem where when I uploaded the .cfg files to the speeder pad all of the files were 0.1kB so I copy and pasted the sontents to their respective folders.  
I have restarted the whole process several times now and I am at a complete loss as to what to do.

**Comment by Baustin525** [comment]
[klippy (3).log](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/files/13387673/klippy.3.log)

**Reply by yjgtfNoname** [reply]
Disculpa a decir verdad soy algo nuevo, estoy teniendo el mismo problema y no tengo idea de por donde empezar, por favor ¿podrias ayudarme?

**Comment by enzoav** [comment]
Can you post a screenshot of the mcu section of your printer.cfg file? I just had the same hiccup and realized I had a typo in the serial line.

**Comment by rmhoward33322** [comment]
Baustin525, I am having the same issue after upgrading my V400 to original Klipper. Were you able to  solve this issue? if yes, what was your solution?
thank you in advance.

**Reply by rmhoward33322** [reply]
I found my issue. Incorrect setting (bootloader offset, set to 8 and should have been 28) when flashing the motherboard firmware.

**Comment by rmhoward33322** [comment]
Here's my klippy log:
[klippy1.log](https://github.com/user-attachments/files/17192582/klippy1.log)
Here's the MCU section in printer.cfg:
########################################
# MCU Settings
########################################

[mcu]
serial: /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0
restart_method: command

[temperature_sensor Speeder_Pad]
sensor_type: temperature_host
min_temp: 10
max_temp: 100

[temperature_sensor Motherboard]
sensor_type: temperature_mcu
min_temp: 0
max_temp: 100

**Reply by daganael** [reply]
I have a similar issue on a V400. after installing the 'Install Official Builds (1 instance)' it can't connect to the MCU. Looking at the logs, and config files, it starts like that : 

serial = /dev/serial/by-id/ < your-mcu-id > (without spaces...)

That's obviously bad but the question is : how do I find my mcu id to fix it ?

**Reply by daganael** [reply]
Found my issue as well when I switched on my brain. sorry guys and many thanks to @Guilouz  for the long and very detail process

**Reply by Jafjas** [reply]
Hi,
Can you tell me your solution?
I am having the same problem. Thanks in advance.

**Reply by daganael** [reply]
Acctually @Guilouz has a page about it under 'Klipper Configurations' https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Get-USB-Serial-from-Motherboard

I missed it at first too
cheers

**Reply by Jafjas** [reply]
Thanks a lot!
I had a lot of issues so I wasn't thinking anymore and just doing hahaha.
